cpp/tests/test_buffer_resource.cpp

```diff
@@ -384,10 +263,9 @@ class BufferResourceCopySliceTest
             0,  // dst_offset
             std::ptrdiff_t(offset)  // src_offset
         );
-
         EXPECT_EQ(slice->mem_type(), dest_type);
-        slice->wait_for_ready();
-        EXPECT_TRUE(slice->is_ready());
+        slice->stream().synchronize();
```

Comment:
*suggestion*: again, we should expose the event so we do `slice->event.host_wait()` here.